src/storage/src/hummock/sstable/forward_sstable_iterator.rs

```diff
@@ -74,20 +74,15 @@ impl SSTableIterator {
         if idx >= self.sst.value().block_count() {
             self.block_iter = None;
         } else {
-            let block = if self.options.prefetch {
-                self.sstable_store
-                    .get_with_prefetch(self.sst.value(), idx as u64, &mut self.stats)
-                    .await?
-            } else {
-                self.sstable_store
-                    .get(
-                        self.sst.value(),
-                        idx as u64,
-                        crate::hummock::CachePolicy::Fill,
-                        &mut self.stats,
-                    )
-                    .await?
-            };
+            let block = self
```

Comment:
If prefetch is done, we already have the cached blocks in `self.sst`. Can we get blocks directly from `self.sst` instead of querying block cache, which has some overhead on lock acqusition and LRU list updates.

Reply:
Also, I think we can avoid the complexity introduced by `TableAccessor` if we use a flag in ReadOption to control whether we need to prefetch or not in the iterator. Why do we prefer `TableAccessor` over a flag in ReadOption?

Reply:
I used another structure `InMemoryIterator` to avoid querying `LRUCache`.

Reply:
I will check whether `blocks` of `TableHolder` is empty to decide whether we need to query `SstableStore`.